packages/query/src/index.ts

```diff
@@ -1129,6 +1129,17 @@ const generateQueryHook = async (
     isQuery = operationQueryOptions.useSuspenseQuery;
   }
 
+  // For non-GET operations, only register query OR mutation hooks, not both
+  let isMutation = verb !== Verbs.GET && override.query.useMutation;
+  if (operationQueryOptions?.useMutation !== undefined) {
+    isMutation = operationQueryOptions.useMutation;
+  }
+
+  // If both query and mutation are true for a non-GET operation, prioritize query
+  if (verb !== Verbs.GET && isQuery && isMutation) {
```

Comment:
If `isQuery` is true, we don't need to consider `isMutation`, right?

```suggestion
  if (verb !== Verbs.GET && isQuery) {
```